src/main/java/bsh/classpath/ClassManagerImpl.java

```diff
@@ -212,15 +212,20 @@ public Class classForName( String name )
         if ( c == null && externalClassLoader != null ) {
             try {
                 c = externalClassLoader.loadClass(name);
+
+                // Don't try any more ClassLoaders if forceExternalClassLoader is true
+                if (forceExternalClassLoader) {
+                    return c;
+                }
             } catch (ClassNotFoundException e) {
-                // fall through
+                // If forceExternalClassLoader is true, there is no reason to return a class here
+                if (forceExternalClassLoader) {
+                    return null;
+                }
             }
         }
 
         // Optionally try context classloader
-        // Note that this might be a security violation
-        // is catching the SecurityException sufficient for all environments?
-        // or do we need a way to turn this off completely?
```

Comment:
I agree with removing the question if we've resolved this, did we resolve this?

Can we add a unit test for this, perhaps to put it behind us.

Reply:
I added the comment again. I agree with you - and I'm not too familiar with bsh, so I can't 100% device whether or not this problem is completely resolved.